frontend-react/docs/proposals/0003-uswds-react-components.md

```diff
@@ -0,0 +1,251 @@
+}
+```
+
+---
+
+<a name="option-two"></a>
+### Option 2: Utilize USWDS Base Library Only
+
+#### Description
+
+Utilize the raw [USWDS library](https://github.com/uswds/uswds) which would remove a level of abstraction to our components. We'd then wrap their components and create a React component that interacts directly with their code which will allow us full customization of props and functionality. CSS will still need to be overridden at times, but can be component-scoped.
+
+#### Pros
+
+- 1:1 relationship with USWDS which means we can upgrade our dependencies without having to rely on Trussworks upgrading theirs.
+- Removes a layer of abstraction so we can create React components with naming conventions and styling that more closely match our designs and syntax.
+- DOM structure will be exposed so we can more easily understand what's going on under the hood which will also help with writing better tests and writing more complex, yet clear, components.
+
+#### Cons
+- Potential breaking changes as we'd be building on top of USWDS.
+- Still need to manually override USWDS styles on occasion using `!important`.
+- Not ideal for very complicated features, like some of the more advanced ideas we had for a ReportStream table.
+
+`RSTable.jsx`:
+
+```js
+import styles from "./RSTable.module.css";
+import classnames from "classnames";
+
+export const RSTable = ({ props, children }) => {
+    // We could easily make a master hash file
+    // with all USWDS CSS classes instead
+    const USWDSTableClasses = {
+      borderless: 'usa-table--borderless',
+      compact: 'usa-table--compact',
+      striped: 'usa-table--striped',
+      stickyHeader: 'rs-table--sticky-header',
+    }
+    
+    // The classnames library recursively flattens arrays
+    const classes = classnames("usa-table", Object.keys(props).map((class) => USWDSTableClasses[class]));
+
+    return (
+      <table className={classes}>
+        {children}
+      </table>
+    );
+};
+```
+
+`RSTable.module.css`:
+
+```css
+.rs-table--sticky-header thead th {
+    position: sticky;
+    top: 0;
+}
+```
+
+---
+
+<a name="option-three"></a>
+### Option 3: Greenstart Customized Components
+
+#### Description
+
+Many of the USWDS components are not very complicated from a Design, animation and overall front-end point of view. Re-creating them from the ground up, especially potentially very complicated components like ReportStream's proposed new tables, would give us granular control and better testing.
+
+#### Pros
+
+- Complete control over presentation and functionality of our components.
+- Can set the standard for other government applications and how they create accessible, usable and beautiful web components for their users.
+- Zero dependencies on external teams to fix broken code or add certain features.
+
+#### Cons
+- Have to maintain, top-to-bottom, the codebase including: testing, accessibility, optimization and everything else that comes with maintaining a library of sorts.
```

Comment:
And having to maintain documentation!  😪

Reply:
Totally! Will add a mention.